test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -333,3 +359,14 @@ func findContainerByName(name string, containers []corev1.Container) *corev1.Con
 
 	return nil
 }
+
+func assertEqualOwnerReference(t *testing.T, resourceType string, resourceNamespacedName types.NamespacedName, ownerReferences []metav1.OwnerReference, expectedOwnerReference metav1.OwnerReference) {
+	assert.Len(t, ownerReferences, 1, fmt.Sprintf("%s %s/%s doesn't have OwnerReferences", resourceType, resourceNamespacedName.Name, resourceNamespacedName.Namespace))
+
+	assert.Equal(t, expectedOwnerReference.APIVersion, ownerReferences[0].APIVersion)
+	assert.Equal(t, "MongoDBCommunity", ownerReferences[0].Kind)
+	assert.Equal(t, expectedOwnerReference.Name, ownerReferences[0].Name)
+	assert.Equal(t, expectedOwnerReference.UID, ownerReferences[0].UID)
+
+	t.Logf("%s %s/%s has the correct OwnerReference!", resourceType, resourceNamespacedName.Name, resourceNamespacedName.Namespace)
```

Comment:
nit: do we need this extra log `assert` functions will report if there are any discrepancies

Reply:
I removed it. You're right, it doesn't add much value (it actually makes the log a bit too verbose)